sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_transport_async.py

```diff
@@ -326,7 +329,7 @@ async def _connect(self, host, port, timeout):
                     # hurray, we established connection
                     return
 
-    def _init_socket(self, socket_settings):
+    def _init_socket(self, socket_settings, read_timeout=1):
         self.sock.settimeout(None)  # set socket back to blocking mode
         self.sock.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
         self._set_socket_options(socket_settings)
```

Comment:
this entire section will be simplified if my PR to use high level streams API is what we end up going with. You don't get access to the underlying socket object, so setting time out explicitly wont be possible. Though Im unsure if it was ever being respected in the first place when the socket is passed in to asyncio.open_connection